claripy/ast/strings.py

```diff
@@ -0,0 +1,169 @@
+    #
+    # TODO: Find a smarter way to do this!
+    STRING_TYPE_IDENTIFIER = 'STRING_'
+    GENERATED_BVS_IDENTIFIER = 'BVS_'
+    MAX_LENGTH = 10000
+
+    def __init__(self, *args, **kwargs):
+        str_len = kwargs['length']
+        kwargs['length'] *= 8
+        super(String, self).__init__(*args, **kwargs)
+        self.string_length = str_len
+
+    def __getitem__(self, rng):
+        if type(rng) is slice:
+            high = rng.start / 8 if rng.start is not None else self.string_length - 1
+            low = rng.stop / 8 if rng.stop is not None else 0
+            if high < 0:
+                high = self.string_length + high
+            if low < 0:
+                low = self.string_length + low
+
+            # Because we are indexing from the end, what was high becomes low and vice-versa
+            high_str_idx = self.string_length - 1 - low
+            low_str_idx = self.string_length - 1 - high
+            return StrExtract(low_str_idx, high_str_idx + 1 - low_str_idx, self)
+        else:
+            raise ValueError("Only slices allowed for string extraction")
+
+    @staticmethod
+    # TODO: Figure out what to convert here
+    #
+    # The use case that i found useful here is during the concretization
+    # of a symbolic address in memory.
+    # When angr has to do this it creates a claripy.If constraints in this form
+    # If(condition, <value_read_if_condition_true>, <value_read_if_condition_false>).
+    # In some case <value_read_if_condition_false> can be MEM_UNINITIALIZED expressed as BVV
+    #
+    # What should we return in this case?
+    def _from_BV(like, value):
+        return value
+
+
+    def replace(self, str_to_replace, replacement):
+        """
+        Replace the first occurence of str_to_replace with replacement
+
+        :param str_to_replace: pattern that has to be replaced
+        :type: String
+        :param replacement: replacement pattern
+        :type : String
+        """
+        return StrReplace(self, str_to_replace, replacement)
+
+    def toInt(self, bitlength):
+        """
+        Convert the string to a bitvector holding the integer
+        representation of the string 
+
+        :param bitlength: size of the biitvector holding the result
+        """
+        return StrToInt(self, bitlength)
+
+    def indexOf(self, pattern, start_idx, bitlength):
+        """
+        Return the start index of the pattern inside the input string in a
+        Bitvector representation, otherwise it returns -1 (always using a BitVector)
+
+        :param bitlength: size of the biitvector holding the result
+        """
+        return StrIndexOf(self, pattern, start_idx, bitlength)
+
+    def raw_to_bv(self):
+        """
+        A counterpart to FP.raw_to_bv - does nothing and returns itself.
+        """
+        if self.symbolic:
+            return BVS(list(self.variables)[0].replace(self.STRING_TYPE_IDENTIFIER, self.GENERATED_BVS_IDENTIFIER), 8)
```

Comment:
I would do `return BVS(next(iter(self.variables)).replace(self.STRING_TYPE_IDENTIFIER, self.GENERATED_BVS_IDENTIFIER), 8)` instead. Also, why is the size `8`?

Reply:
I think that was an hack that Thorsten implemented for one of his test case... Now I found the source of the bug and I patched it.